packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -369,6 +381,71 @@ export default createRule<Options, MessageId>({
       }
     }
 
+    function checkClassLikeOrInterfaceNode(
+      node:
+        | TSESTree.ClassDeclaration
+        | TSESTree.ClassExpression
+        | TSESTree.TSInterfaceDeclaration,
+    ): void {
+      const tsNode = services.esTreeNodeToTSNodeMap.get(node);
+
+      const heritageTypes = getHeritageTypes(checker, tsNode);
+      if (!heritageTypes?.length) {
+        return;
+      }
+
+      for (const nodeMember of tsNode.members) {
+        const memberName = nodeMember.name?.getText();
+        if (memberName === undefined) {
+          // Call/construct/index signatures don't have names. TS allows call signatures to mismatch,
+          // and construct signatures can't be async.
+          // TODO - Once we're able to use `checker.isTypeAssignableTo` (v8), we can check an index
```

Comment:
v8 is upon us! Though of course this PR has been in the works since long before that. 

@alythobani, let's merge as-is, but would you mind filing a followup (that we now know will not be blocked 🙂)

Reply:
Sweet, yeah definitely! Do you mean filing an issue for this? And I can start looking into a proper implementation too if you'd like